pkg/target/manager.go

```diff
@@ -282,8 +292,122 @@ func (m *managerImpl) validateSeed(ctx context.Context, seed *gardencorev1beta1.
 	return nil
 }
 
+func domainsFromConfiguration(m *managerImpl) (map[string]string, map[string]string, error) {
+	domainRegexp, err := regexp.Compile(`(http://|https://)?(.+\..+)`)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to compile regular expression: %v", err)
+	}
+
+	dashboardDomains := make(map[string]string)
+	shootDomains := make(map[string]string)
+
+	for _, g := range m.Configuration().Gardens {
+		match := domainRegexp.FindStringSubmatch(g.DashboardDomain)
+		if match != nil {
+			dashboardDomains[g.Name] = match[2]
+		}
+
+		match = domainRegexp.FindStringSubmatch(g.ShootDomain)
+		if match != nil {
+			shootDomains[g.Name] = match[2]
+		}
+	}
+
+	return dashboardDomains, shootDomains, nil
+}
+
+func matchGardenDomain(domains map[string]string, domain string) (string, string) {
+	for garden, gDomain := range domains {
+		if strings.Contains(domain, gDomain) {
+			return garden, domain
+		}
+	}
+
+	return "", ""
+}
+
+func targetFlagsFromDomain(ctx context.Context, m *managerImpl, t *targetImpl, name string) (TargetFlags, error) {
```

Comment:
This should be a method of the managerImp. It should return a Target and not a TargetFlags. Why does it have in input parameter of type targteImpl. There is no need to depend on an impl of an interface. The only info we use is the gardenName.